### PR TITLE
Auto Key Cipher in C, C++ and Java

### LINKED ISSUE #4443: Auto Key Cipher in C, C++ and Java
**This is a(n):**

- [x] New algorithm
- [x] Update to an existing algorithm
- [ ] Error
- [x] Proposal to the Repository

**Details:**

 Implementation of this algorithm will done in C, C++ an Java. Please assign this to me  @AdiChat

### PR DESCRIPTION
**Pull request by ravireddy07**
**Fixes issue:**

fixes #4443 

**Changes:**

-- Added Auto Key Cipher in C.
-- Added Auto Key Cipher in C++.
-- Added Auto Key Cipher in Java.

### CODE REVIEW

File: code/cryptography/src/autokey_cipher/autokey_cipher.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+
+public class autokey
+{
+	public static String createkey(String plaintext, int keycount, HashMap<Character, Integer> hashmap1, HashMap<Integer, Character> hashmap2)
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey_cipher.java]: Fix indentation in this file to 4 spaces.

File: code/cryptography/src/autokey_cipher/autokey_cipher.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
+#include <iostream>
+#include <string.h>
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey_cipher.cpp]: I don't think this header is used anywhere.

File: code/cryptography/src/autokey_cipher/autokey_cipher.c [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        if (isalpha((*text)[i]))
+        {
+            (*text)[i] = toupper((*text)[i]);
+        }
+    }
+
+    int nextkey, keyvalue, result;
+    nextkey = toupper(key) - 'A';
+
+    for (int i = 0; i < textsize; i++)
```
Comment (arnavb) [inline code/cryptography/src/autokey_cipher/autokey_cipher.c]: In all loops in C/C++, use pre-increment.

File: code/cryptography/src/autokey_cipher/autokey_cipher.java (hunk 1 above)
Comment (ravireddy07) [reply code/cryptography/src/autokey_cipher/autokey_cipher.java]: indentation is used clang-format(x64)

**Comment by ravireddy07** [comment]
changes made, check once @arnavb

File: code/cryptography/src/autokey_cipher/autokey_cipher.java (hunk 1 above)
Comment (arnavb) [reply code/cryptography/src/autokey_cipher/autokey_cipher.java]: @ravireddy07 That was only for C/C++. For Java we use 4 space indentation.

Comment (ravireddy07) [reply code/cryptography/src/autokey_cipher/autokey_cipher.java]: Code indented. please check now. @arnavb

Comment (ravireddy07) [reply code/cryptography/src/autokey_cipher/autokey_cipher.java]: Indented the code on github only. 4-space Indenattion.

**Comment by ravireddy07** [comment]
@arnavb  Sir, please add corresponding labels for this PR or Issue. so that it will be added/counted towards my contribution points.

**Comment by arnavb** [comment]
@ravireddy07 Done for the issue.

**Pull request merged**